src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,69 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(false);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    if (currentTodo) {
+      setLoading(true);
+      getUser(currentTodo?.userId)
+        .then(setUser)
+        .finally(() => setLoading(false));
+    }
+  }, [currentTodo]);
+
   return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
-
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+    <>
+      {currentTodo && (
+        <div className="modal is-active" data-cy="modal">
+          <div className="modal-background" />
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+          {loading && <Loader />}
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+          <div className="modal-card">
+            <header className="modal-card-head">
+              <div
+                className="modal-card-title has-text-weight-medium"
+                data-cy="modal-header"
+              >
+                Todo #{currentTodo.id}
+              </div>
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                type="button"
+                className="delete"
+                data-cy="modal-close"
+                onClick={() => dispatch({ type: 'currentTodo/REMOVE' })}
```

Comment:
In this case, you need a handler. It will be more readable. ( also, the best way to add payload to avoid side effects. For example: when you try to call action.payload)

Reply:
Please clarify what do you mean?)